from_dict/_from_dict.py

```diff
@@ -63,76 +62,86 @@ def __eq__(self, o: object) -> bool:
 
 
 if IS_GE_PYTHON38:
-
-    def get_origin(t):
-        return typing.get_origin(t)
-
-    def get_args(t):
-        return typing.get_args(t)
-
+    from typing import get_origin, get_args
 else:
 
-    def get_origin(t):
-        if hasattr(t, "__origin__"):
-            return t.__origin__
+    def get_origin(tp) -> Optional[type]:
+        if hasattr(tp, "__origin__"):
+            return tp.__origin__
         else:
             return None
 
-    def get_args(t):
-        if hasattr(t, "__args__"):
-            return t.__args__
+    def get_args(tp) -> tuple:
+        if hasattr(tp, "__args__"):
+            return tp.__args__
         else:
-            return None
+            return ()
 
 
-def type_check(v, t) -> None:
+def type_check(check_stack: list, v: Any, t: type) -> None:
     """Raise RuntimeTypeError if given value does not agree with given type"""
     # This uses typing.get_args and typing.get_origin
+    def location():
```

Comment:
Same here, it seems there's an improvement to error messages and a functional change. Could we separate those into separate commits?